Foundation/FileManager.swift

```diff
@@ -64,18 +64,320 @@ open class FileManager : NSObject {
+            
+        case .musicDirectory:
+            guard domain == .user else { return [] }
+            return [ URL(fileURLWithPath: "Music", isDirectory: true, relativeTo: URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true)) ]
+
+        case .picturesDirectory:
+            guard domain == .user else { return [] }
+            return [ URL(fileURLWithPath: "Pictures", isDirectory: true, relativeTo: URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true)) ]
+
+        case .sharedPublicDirectory:
+            guard domain == .user else { return [] }
+            return [ URL(fileURLWithPath: "Public", isDirectory: true, relativeTo: URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true)) ]
+
+        case .trashDirectory:
+            let userTrashURL = URL(fileURLWithPath: ".Trash", isDirectory: true, relativeTo: URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true))
+            if domain == .user || domain == .local {
+                return [ userTrashURL ]
+            } else {
+                return []
+            }
+            
+        case .itemReplacementDirectory:
+            // This directory is only returned by url(for:in:appropriateFor:create:)
+            return []
+        }
+        #elseif !os(Windows)
+        // FHS/XDG-compliant OSes:
+        switch directory {
+        case .autosavedInformationDirectory:
+            let runtimePath = _SwiftValue.fetch(nonOptional: _CFXDGCreateDataHomePath()) as! String
+            return [ URL(fileURLWithPath: "Autosave Information", isDirectory: true, relativeTo: URL(fileURLWithPath: runtimePath, isDirectory: true)) ]
+            
+        case .desktopDirectory:
+            guard domain == .user else { return [] }
+            return [ _XDGUserDirectory.desktop.url ]
+            
+        case .documentDirectory:
+            guard domain == .user else { return [] }
+            return [ _XDGUserDirectory.documents.url ]
+            
+        case .cachesDirectory:
+            guard domain == .user else { return [] }
+            let path = _SwiftValue.fetch(nonOptional: _CFXDGCreateCacheDirectoryPath()) as! String
+            return [ URL(fileURLWithPath: path, isDirectory: true) ]
+            
+        case .applicationSupportDirectory:
+            guard domain == .user else { return [] }
+            let path = _SwiftValue.fetch(nonOptional: _CFXDGCreateDataHomePath()) as! String
+            return [ URL(fileURLWithPath: path, isDirectory: true) ]
+
+        case .downloadsDirectory:
+            guard domain == .user else { return [] }
+            return [ _XDGUserDirectory.download.url ]
+            
+        case .userDirectory:
+            guard domain == .local else { return [] }
+            return [ URL(fileURLWithPath: "/home", isDirectory: true) ]
+            
+        case .moviesDirectory:
+            return []
```

Comment:
Could we use `_XDGUserDirectory.videos.url` here?

Reply:
Good catch.